### ISSUE TITLE
Get loss per image, with a trained model 

### ISSUE THREAD

**Issue by NicoJorgensen1** [ultralytics/ultralytics#8015]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### What I want to achieve
I want to obtain a mapping like `path2loss = {img_path: loss}`. I am using ultralytics==8.0.186 (due to wandb integration). I use yolov8 for object detection.

### How I thought I could achive that
To get a mapping similar to what I need, I need a way to compute the loss per image with a trained model. I can do something like 
`from ultralytics import YOLO ` 
`model = YOLO(model=path/to/weights)`
`pred = model(path/to/image)`
`from some_ultralytics_script import some_loss_func`
`loss = some_loss_func(pred, some_target_bbox(es)`
However, I cannot seem to make this work. And neither model.train/val/predict natively returns the loss. 

### What I have tried so far
When I look into the loss.py (from ultralytics/yolo/utils/loss.py) I can't seem to how to use that script. 
When debugging my way through `model.train()`, I can see that I end up in the class `v8DetectionLoss` (in /usr/local/lib/python3.10/dist-packages/ultralytics/utils/loss.py). The `__call__` function of this class takes predictions and a batch. 
However, at this point the batch has been altered a lot by some preprocessing steps. I can't seem to wrap my head round what happens with the input and the predictions up untill this point.

 

I hope that someone can please guide me in the right direction as to how I can most "easily" obtain my wanted path2loss mapping.

**Comment by fiksii-copilot** [comment]
@NicoJorgensen1 If you find the solution, please share it
